src/zotero-ocr.js

```diff
@@ -0,0 +1,324 @@
+                contents = contents.replace(/(?:\r\n|\r|\n)/g, '<br />');
+                let newNote = new Zotero.Item('note');
+                newNote.setNote(contents);
+                newNote.parentID = item.id;
+                await newNote.saveTx();
+            }
+            
+            if (Zotero.Prefs.get("zoteroocr.outputHocr")) {
+                let contents = await Zotero.File.getContentsAsync(ocrbase + '.hocr');
+                // replace the absolute paths of images with relative ones
+                let escapedDir = dir.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+                let regexp = new RegExp(escapedDir + "/", 'g');
+                contents = contents.replace(regexp, '');
+                // split content into the preamble and pages
+                contents = contents.replace("</body>\n</html>", '');
+                let parts = contents.split("<div class='ocr_page'");
+                let preamble = parts[0];
+                // create new html attachments including hocrjs for individual pages
+                let maximumPagesAsHtml = parseInt(Zotero.Prefs.get("zoteroocr.maximumPagesAsHtml"));
+                let upperLimit = parts.length;
+                if (!(isNaN(maximumPagesAsHtml)) && (maximumPagesAsHtml + 1 < upperLimit)) {
+                    upperLimit = maximumPagesAsHtml + 1;
+                }
+                for (let i = 1; i < upperLimit; i++) {
+                    let pagename = 'page-' + i + '.html';
+                    let htmlfile = Zotero.File.pathToFile(OS.Path.join(dir, pagename));
+                    let pagecontent = preamble + "<div class='ocr_page'" + parts[i] +   '<script src="https://unpkg.com/hocrjs"></script>\n</body>\n</html>';
+                    Zotero.File.putContents(htmlfile, pagecontent);
+                    await Zotero.Attachments.linkFromFile({
+                        file: OS.Path.join(dir, pagename),
+                        contentType: "text/html",
+                        parentItemID: item.id
+                    });
+                }
+            }
+
+            // attach PDF if it is a new one
+            if (Zotero.Prefs.get("zoteroocr.outputPDF") && !(Zotero.Prefs.get("zoteroocr.overwritePDF"))) {
+                await Zotero.Attachments.linkFromFile({
+                    file: ocrbase + '.pdf',
+                    parentItemID: item.id
+                });
+            }
+            
+            if (!Zotero.Prefs.get("zoteroocr.outputPNG") && imageListArray) {
+                // delete image list
+                await Zotero.File.removeIfExists(imageList);
+                // delete PNGs
+                for (let imageName of imageListArray) {
+                    await Zotero.File.removeIfExists(imageName);
+                }
+            }
+        }
+    },
+    
+    async main() {
+        // Global properties are included automatically in Zotero 7
+        var host = new URL('https://foo.com/path').host;
+        
+    },
```

Comment:
```suggestion
```

Reply:
`main` is a relict from copy+paste which is not needed here.

Reply:
I had assumed the function was necessary (although perhaps the content might not do anything useful), but if it isn't, of course let's get rid of it.